ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/folsom/CompressionTranscoder.java

```diff
@@ -0,0 +1,32 @@
+package com.outbrain.ob1k.cache.memcache.folsom;
+
+import com.outbrain.ob1k.cache.memcache.compression.*;
+import com.spotify.folsom.Transcoder;
+
+import java.util.Objects;
+
+/**
+ * Created by bshushi on 26/03/2017.
+ */
+public class CompressionTranscoder<T> implements Transcoder<T> {
+
+  private final Transcoder<T> delegate;
+  private final Compressor compressor;
+
+  public CompressionTranscoder(final Transcoder<T> delegate, Compressor compressor) {
+    this.delegate = Objects.requireNonNull(delegate);
+    this.compressor = compressor;
+  }
+
+  @Override
+  public T decode(byte[] bytes) {
+    Objects.requireNonNull(bytes);
+    byte[] decompress = compressor.decompress(bytes);
```

Comment:
code style thing: please add final for all defined variables (and method arguments)